lib/internal/http2/core.js

```diff
@@ -2526,6 +2527,8 @@ function connectionListener(socket) {
                  'If this is a HTTP request: The server was not ' +
                  'configured with the `allowHTTP1` option or a ' +
                  'listener for the `unknownProtocol` event.\n');
+
+      socket.on('error', () => {});
```

Comment:
In addition to the other feedback, could we abstract this into `const noop = () => {};` somewhere at the top level? Perhaps close to the `emit` helper.